### DISCUSSION TITLE
Umi & Bigfish 想法征集 02

### DISCUSSION THREAD

**Discussion by sorrycc** [umijs/umi#10568] [Ideas]
> Why this issue? 计划在 Umi Contributor 群增加一条规则。每人每周贡献 2 个改进 Umi 或 Bigfish 的想法（辟起、辟殊、云谦），外包同学减半（1 个），外部 Contributor 随意，我每周三会在群里发个 issue 做收集。

直接回复即可。（含内部敏感信息的可以私发我）

**Comment by PeachScript** [comment]
- caniuse-lite 彻底解，现在启动 CLI 又有 outdated 的 warning 了，去年的解法是预打包产物里注释掉警告，但没有彻底解 outdated 的问题
- Bigfish 4 仓库 examples 下不是用的 workspace 的 bigfish 而是单独装了一个，要用 `../../bin/bigfish` 调用对共建者来说有调试成本，配成 workspace 又会触发强约束，得对 examples 加特判？
- 小问题：
  - `lessLoader.modifyVars` 会覆盖 `theme` 配置项，对普通用户来说这是个隐式行为，对插件开发者来说，缺少一个标准化的接口增加 Less 变量

**Reply by sorrycc** [reply]
1、问题 1 是 umi 4 吗？我试了下 umi 4 的简单场景没有复现。
<img width="1145" alt="image" src="https://user-images.githubusercontent.com/35128/222018849-6750aa98-01ad-4de0-9211-e2138b4507b1.png">
2、问题 2 已记录，排在下周解

**Comment by sorrycc** [comment]
1、Bigfish 在 dev 和 build 报错时增加引导用户使用 doctor
2、新增一个功能，比如代码里检测到 styled 时，自动增加相应的 import 语句，不知能否做成 prettier 规则
3、改进 Umi 的类型提示，参考下 Nuxt
4、Umi UI 重启，哈哈（已完成第一步）
5、Umi 添加 plugins 这个操作可以进一步简化，1）只要在 plugins 里添加模块，依赖部分可以自动安装，2）内置插件快速添加，无需加 @umijs/plugins/dist 前缀，3）支持函数。
6、GPTIndex x Umi

**Comment by xierenyuan** [comment]
- Umi 仓库  umi 相关依赖使用 `workspace:* ` 代替减少开发冲突
- Umi ui 引入 https://github.com/modern-js-dev/webpack-inspector 或类似的能力(可以是直接包含 或者基于这个开发)
- 是不是可以做本地的组件预览类似 https://github.com/johnsoncodehk/vite-plugin-vue-component-preview
- mfsu 增加 sourcemap 的能力, 现在开启 mfsu 后三方报错 调试困难

**Reply by stormslowly** [reply]
第 4 点：很有难度建议一起搞起来。

**Reply by sorrycc** [reply]
@stormslowly 难点在哪？实现还是速度？

**Reply by sorrycc** [reply]
1、DONE
2、感觉可以做

**Comment by stormslowly** [comment]
1. Umi 文档 提供文档GPT 功能。参考 https://github.com/arc53/DocsGPT
2. Umi 增加  telemetry 功能，了解 Umi 实际运行情况 参考 https://github.com/vercel/next.js/tree/canary/packages/next/src/telemetry 
3. Bigfish checker 执行报告增加 UI 页面，方便查看错误

**Reply by hanzebang** [reply]
> 3. Bigfish checker 执行报告增加 UI 页面，方便查看错误

gkd，让俺也学习一下，doctor 也加一个 UI 🤗

**Reply by sorrycc** [reply]
1、上周跑了个 POC，基于 GPT Index
2、现在就有部分，但没那么细，有些数据需要补充下，比如 dev、build 的时间等，但感觉是否也不需要那么细？

**Comment by hanzebang** [comment]
1. Umi/Bigfish （也许在api）添加一些 error 勾子，onError、onServiceError、onCompilingError 等，可以做一些错误引导或反馈等统一处理；
2. 如何让用户学会“庖丁解牛”，问我们排查后真正跟 Bigfish 相关的问题，而不是直接把当时的大环境整个都扔给我们，要我们一点一点排查
  - 比如说 from 'Test' 不能自动引入 'Test/index'，(Bigfish demo 是 ok 的，大概率是 wf 插件或者其他环境问题导致的)
  - 比如跨域问题，后端限制的跨域总想让我们解决
  - 比如网页 title 刚开始正常，后面显示 url 的问题，主应用+子应用+其他脚本，我要挨个 js 去找是谁干的
  - 等等

**Reply by sorrycc** [reply]
1、记录了
2、源头是「报错信息」，建议从这个源头开始设计文档、doctor 和框架层的引导，空了可以往深了想想怎么做。比如跨域问题和 title 问题都可以有一篇汇总性的文档，把各个分支都描述清楚，后续有 GPT 辅助答疑之后也需要这些文档；持续完善 doctor，然后要求用户提问前先跑 doctor，或者我们通过 cloudide 或 tr 自动给他跑个 doctor 的报告。

**Comment by wyy0512** [comment]
如何让用户先自查（需要小蜜配合的）

- 咨询类问题可以在用户进入人工服务时让机器人自动贴上 bigfish 官网还有各个与之相关的文档入口（因为有碰到过在官网有的文档，还是会有人来询问有没有某某文档）。

- 报错类：可以放些自我排查的基础性指导文档（雨燕构建报错、控制台报错、项目启动终端报错等），也是用户进入人工服务时，机器人先抛给用户让其先自查，查不出来值班人员再介入。
